File: web/src/api/searchResult.js
import service from '@/utils/request'

export const createSearchResult = (data) => {
     return service({
         url: "/searchResult/createSearchResult",
         method: 'post',
         data
     })
 }

 export const deleteSearchResult = (data) => {
     return service({
         url: "/searchResult/deleteSearchResult",
         method: 'delete',
         data
     })
 }

 export const deleteSearchResultByIds = (data) => {
     return service({
         url: "/searchResult/deleteSearchResultByIds",
         method: 'delete',
         data
     })
 }

 export const updateSearchResult = (data) => {
     return service({
         url: "/searchResult/updateSearchResult",
         method: 'post',
         data
     })
 }

 export const findSearchResult = (params) => {
     return service({
         url: "/searchResult/findSearchResult",
         method: 'get',
         params
     })
 }

 export const getSearchResultList = (params) => {
     return service({
         url: "/searchResult/getSearchResultList",
         method: 'get',
         params
     })
 }

export const exportSearchResult = async (params) => {
    try {
        const response = await service({
            url: '/searchResult/exportSearchResult',
            method: 'get',
            params,
            responseType: 'blob',  // Set response type to blob
            headers: {
                'Accept': 'text/csv'
            }
        });
        const contentDisposition = response.headers?.['content-disposition'] || response.headers?.get('content-disposition');
        const filename = contentDisposition
            ? contentDisposition.split('filename=')[1]?.replace(/"/g, '')
            : 'export.csv';
        const blob = new Blob([response.data], {
            type: 'text/csv;charset=utf-8;'
        });
        const downloadUrl = window.URL.createObjectURL(blob);

        const link = document.createElement('a');
        link.href = downloadUrl;
        link.download = filename;
        document.body.appendChild(link);
        link.click();

        document.body.removeChild(link);
        window.URL.revokeObjectURL(downloadUrl);
    } catch (error) {
        console.error('Export failed:', error);
        throw error; // Propagate error to caller
    }
};

 export const updateSearchResultStatusByIds = (data) => {
     return service({
         url: '/searchResult/updateSearchResultStatusByIds',
         method: 'post',
         data
     })
 }

 export const startFilterTask = () => {
    return service({
        url: "/searchResult/startSecFilterTask",
        method: "post"
    })
 }

 export const getTaskStatus = () => {
    return service({
        url: "/searchResult/getTaskStatus",
        method: "get"
    })
 }

File: server/router/search_result.go
package router

import (
	"github.com/gin-gonic/gin"
	"github.com/madneal/gshark/api"
	"github.com/madneal/gshark/middleware"
)

func InitSearchResultRouter(Router *gin.RouterGroup) {
	SearchResultRouter := Router.Group("searchResult").Use(middleware.OperationRecord())
	{
		SearchResultRouter.POST("createSearchResult", api.CreateSearchResult)
		SearchResultRouter.DELETE("deleteSearchResult", api.DeleteSearchResult)
		SearchResultRouter.DELETE("deleteSearchResultByIds", api.DeleteSearchResultByIds)
		SearchResultRouter.POST("updateSearchResult", api.UpdateSearchResult)
		SearchResultRouter.GET("findSearchResult", api.FindSearchResult)
		SearchResultRouter.GET("getSearchResultList", api.GetSearchResultList)
		SearchResultRouter.GET("exportSearchResult", api.ExportSearchResult)
		SearchResultRouter.POST("updateSearchResultStatusByIds", api.UpdateSearchResultByIds)
		SearchResultRouter.POST("startSecFilterTask", api.StartSecFilterTask)
		SearchResultRouter.GET("getTaskStatus", api.GetTaskStatus)
		SearchResultRouter.POST("startAITask", api.StartAITask)
	}
}
